cmd/minishift/cmd/delete.go

```diff
@@ -36,6 +37,9 @@ var deleteCmd = &cobra.Command{
 }
 
 func runDelete(cmd *cobra.Command, args []string) {
+	instancedata.Instance().ReadInstanceData()
```

Comment:
So, I guess libmachine uses just the existence of the machine directory as criteria to determine whether there should be an existing VM - https://github.com/docker/machine/blob/master/libmachine/persist/filestore.go#L100. A bit weak imo. Anyways, in this case we need to manage our own instance specific directory aside from lib machine and making sure it gets cleaned up on _delete_. Open to any directory layout suggestions. Maybe it is a good thing anyways to stay of off libmachine's managed directory.